### PR TITLE
Profiles: Add Backgrounds, Music, Nature, and Type

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

File: wavelength-plugins/exp.js [hunk 1]
```diff
@@ -108,6 +108,18 @@ function addExp(user, room, amount) {
 					Monitor.adminlog(user.userid + ' has earned a chatroom for reaching level ' + level + '!');
 					reward = 'a Chatroom. To claim your chatroom, Contact a Leader (&) or Administrator (~).';
 					break;
+				case 45:
```
Comment (Lycanium-Z) [inline wavelength-plugins/exp.js]: 45 and 50 feels too high for this tbh

Comment (DeathlyPlays) [reply wavelength-plugins/exp.js]: tbh I was just setting it in the ongoing increments until HoeenHero changes it

Comment (HoeenCoder) [reply wavelength-plugins/exp.js]: Id agree, custom symbol is stupid so maybe just allow both of these for level 5?

**Comment by DeathlyPlays** [comment]
Perhaps later

**Comment by DeathlyPlays** [comment]
@HoeenCoder bump

File: wavelength-plugins/exp.js [hunk 2]
```diff
@@ -108,6 +109,12 @@ function addExp(user, room, amount) {
 					Monitor.adminlog(user.userid + ' has earned a chatroom for reaching level ' + level + '!');
 					reward = 'a Chatroom. To claim your chatroom, Contact a Leader (&) or Administrator (~).';
 					break;
+				case 50:
+					Economy.logTransaction(user.userid + ' received profile music for reaching level ' + level + '.');
+					WL.messageSeniorStaff(user.userid + ' has earned profile music for reaching level ' + level + '.');
+					Monitor.log(user.userid + ' has earned profile music for reaching level ' + level + '!');
+					reward = 'Profile Music. To claim your profile music, contact a Global Staff (% and up).';
+					break;
```
Comment (HoeenCoder) [inline wavelength-plugins/exp.js]: This is supposed to be 5?

File: wavelength-plugins/exp.js [hunk 3]
```diff
@@ -64,9 +64,10 @@ function addExp(user, room, amount) {
 				let reward = '';
 				switch (level) {
 				case 5:
-					Economy.logTransaction(user.userid + ' received a custom symbol for reaching level ' + level + '.');
-					user.canCustomSymbol = true;
-					reward = 'a Custom Symbol. To claim your custom symbol, use the command /customsymbol [symbol]';
+					Economy.logTransaction(user.userid + ' received a profile background and profile music for reaching level ' + level + '.');
+					WL.messageSeniorStaff(user.userid + ' has earned a profile background and profile music for reaching level ' + level + '.');
```
Comment (HoeenCoder) [inline wavelength-plugins/exp.js]: Instead of PMing upper staff, lets just log to staff (which we do below).

Comment (DeathlyPlays) [reply wavelength-plugins/exp.js]: Ok

File: wavelength-plugins/exp.js (hunk 2 above)
Comment (DeathlyPlays) [reply wavelength-plugins/exp.js]: Whoops

File: wavelength-plugins/exp.js [hunk 4]
```diff
@@ -64,9 +64,9 @@ function addExp(user, room, amount) {
 				let reward = '';
 				switch (level) {
 				case 5:
-					Economy.logTransaction(user.userid + ' received a custom symbol for reaching level ' + level + '.');
-					user.canCustomSymbol = true;
-					reward = 'a Custom Symbol. To claim your custom symbol, use the command /customsymbol [symbol]';
+					Economy.logTransaction(user.userid + ' received a profile background and profile music for reaching level ' + level + '.');
+					Monitor.log(user.userid + ' has earned a profile background and profile music for reaching level ' + level + '!');
+					reward = 'a Profile Background and Profile Music. To claim your profile background and profile music, contact a Global Staff (% and up).';
```
Comment (HoeenCoder) [inline wavelength-plugins/exp.js]: "contact a global staff member"

**Comment by DeathlyPlays** [comment]
@HoeenCoder bump

**Comment by DeathlyPlays** [comment]
@HoeenCoder this is tested yknow

**Comment by HoeenCoder** [comment]
I want to test it myself anyways because often times i find your code dosent work when you test it :/

**Comment by DeathlyPlays** [comment]
:(

**Comment by DeathlyPlays** [comment]
@HoeenCoder bump bump bump
Bumpity bump
Ba dum bump bump
Bump da bump ba da duh
Bump bump bump
@Lycanium-Z cue chorus

**Comment by HoeenCoder** [comment]
/me closes PR

Someone else test this, I dont have time.

**Comment by DeathlyPlays** [comment]
Like it works on my server LOL

**Comment by HoeenCoder** [comment]
Often times when you say it works or its tested, i find its not when i run tests. How do you test things anyways?

**Comment by DeathlyPlays** [comment]
Commands work & such >_>

**Comment by HoeenCoder** [comment]
work as in they dont crash or dont crash AND work properly? (also checking things like /profile in this case)

**Comment by DeathlyPlays** [comment]
Don't crash + work

**Comment by DeathlyPlays** [comment]
Feel free to test yourself smh

**Comment by DeathlyPlays** [comment]
@Mystifi @HoeenCoder @Lycanium-Z @Arrays-Arr pinging for final reviews test runs

File: wavelength-plugins/profile.js [hunk 5]
```diff
@@ -275,6 +282,81 @@ exports.commands = {
 		},
 	},
 
+	bg: 'background',
+	background: {
+		set: 'setbg',
+		setbackground: 'setbg',
+		setbg: function (target, room, user) {
+			if (!this.can('broadcast')) return false;
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: shouldnt this be lock?

File: wavelength-plugins/profile.js [hunk 6]
```diff
@@ -343,18 +484,37 @@ exports.commands = {
 			return '<img src="http://flags.fmcdn.net/data/flags/normal/' + ip.country.toLowerCase() + '.png" alt="' + ip.country + '" title="' + ip.country + '" width="20" height="10">';
 		}
 
+		function background(buddy) {
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: lets use a variable name that makes more sense like `user`

File: wavelength-plugins/profile.js [hunk 7]
```diff
@@ -343,18 +484,37 @@ exports.commands = {
...
+			let bg = Db.backgrounds.get(buddy);
+			if (!Db.backgrounds.has(buddy)) return '<div>';
+			return '<div style="background:url(' + bg + ')">';
+		}
+
+		function song(fren) {
```
Comment (HoeenCoder) [inline wavelength-plugins/profile.js]: lets use a variable name that makes more sense like `user`

**Comment by HoeenCoder** [comment]
Alright fine, if you say its tested, and you actually did it, ill merge since i lack time to do it, and audio restrictions :p

File: wavelength-plugins/profile.js (hunk 5 above)
Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: oh my bad LOL that's what we use on my server

File: wavelength-plugins/profile.js (hunk 6 above)
Comment (DeathlyPlays) [reply wavelength-plugins/profile.js]: Sure thing

**Pull request merged**